genetic_algorithms/problems/traveling_salesman_problem/move_generator.py

```diff
@@ -0,0 +1,28 @@
+import random
+from itertools import combinations
+
+
+from genetic_algorithms.models.move_generator import MoveGenerator
+from genetic_algorithms.problems.traveling_salesman_problem.state import TravelingSalesmanState
+from genetic_algorithms.problems.traveling_salesman_problem.moves import SwapEdges
+from genetic_algorithms.problems.traveling_salesman_problem.models.edge import Edge
+from typing import Generator
+
+
+class TravelingSalesmanMoveGenerator(MoveGenerator):
+    def __init__(self, depot_idx: int):
+        self._depot_idx = depot_idx
+
+    def _is_depot_start(self, a: Edge):
+        return a.start != self._depot_idx
+
+    def _is_depot_end(self, b: Edge):
+        return b.end != self._depot_idx
+
+    def random_move(self, state: TravelingSalesmanState) -> SwapEdges:
+        return SwapEdges(state, random.choice(list(state.edges)), random.choice(list(state.edges)))
```

Comment:
Nie uwzględniasz ograniczeń. 
W tej chwili może być tak, że wygenerujesz ruch który przestawi `depot` z początku na środek.


Ja bym zrobił to jakoś tak:
```python
generate_move = lambda: SwapEdges(state, random.choice(list(state.edges)), random.choice(list(state.edges)))
random_move = generate_move()
while not self._is_move_satisfies_constraints(random_move):
   random_move = generate_move()
```
Gdzie `_is_move_satisfies_constraints` - to funkcja sprawdzająca ograniczenia na problemie.

Reply:
A no właśnie dlatego napisałam komentarz, że jest źle - nie chodziło mi o sam generator ruchów, tylko że ten random_move nie uwzględniał ograniczeń. Ale tu widzę, że spoko rozwiązanie podałeś

Reply:
Dalej chyba tego nie dodałaś